lib/compress/zstd_fast.c

```diff
@@ -288,9 +305,7 @@ ZSTD_compressBlock_fast_noDict_generic(
     anchor = ip0;
 
     /* write next hash table entry */
-    if (ip1 < ip0) {
-        hashTable[hash1] = (U32)(ip1 - base);
-    }
+    hashTable[hash1] = (U32)(ip1 - base);
```

Comment:
If I understand correctly, when `step > 4`, then `hashTable[hash1] = 0;`, so it's removing a potentially useful past reference from the hash table ?

Reply:
Correct. So this does affect levels -5 and beyond. But otherwise it has basically no effect, since that situation is very rare otherwise.

Reply:
What would be the consequences of _not_ having this hash table write operation ?

Reply:
So I thought about this some more and came up with a better way to do it!

To answer your question, we would gain some speed and lose some ratio. Here's a quick benchmark comparison:

Comp | Version | Speed | Ratio
--|--|--|--
gcc-11 | no write | 348.2 | x2.876
gcc-11 | `dev` | 345.7 | x2.887
gcc-11 | this PR | 348.7 | x2.887
clang-13 | no write | 346.1 | x2.876
clang-13 | `dev` | 343.9 | x2.887
clang-13 | this PR | 348.5 | x2.887